std/algorithm/sorting.d

```diff
@@ -1548,14 +1662,14 @@ sort(alias less = "a < b", SwapStrategy ss = SwapStrategy.unstable,
 private void quickSortImpl(alias less, Range)(Range r, size_t depth)
 {
     import std.algorithm.mutation : swap, swapAt;
-    import std.algorithm.comparison : min;
+    import std.algorithm.comparison : min, max;
 
     alias Elem = ElementType!(Range);
-    enum size_t optimisticInsertionSortGetsBetter = 25;
-    static assert(optimisticInsertionSortGetsBetter >= 1);
+    enum size_t shortSortGetsBetter = max(32, 1024 / Elem.sizeof);
```

Comment:
How did you find that threshold and the 1024/ElementSize?
Any benchmarks to find optimal thresholds?